Removed strict validation of external volume name (#7023)
mesosphere/marathon: changelog.md

For more detailed information, see the JIRA ticket [MARATHON-8663](https://jira.mesosphere.com/browse/MARATHON-8663)

## Changes from 1.8.218 to 1.8.xxx

### External Volume Validation changes

#### Relaxed name validation

As there are some external volume providers which require options in the volume name, the strict validation of the name on the external volume is now removed.

As the uniqueness check is based on the volume name, this may lead to some inconsistencies, for the sake of uniquenes, the following volumes are distinct:

```json
"volumes": [
      {
        "external": {
          "name": "name=volumename,option1=value",
        },
      }
    ],
```

```json
"volumes": [
      {
        "external": {
          "name": "option1=value,name=volumename",
        },
      }
    ],
```

#### Optional uniqueness check

> Additionally, there is a validation that prevents an external volume with the same name to be used on multiple apps. Multiple external volume providers now

Reviewer: ```suggestion
Previously, Marathon would validate that an external volume with the same name is only used once across multiple apps. This was due to the initial implementation being focused on Rexray+EBS. However, multiple external volume providers now
```

> allow shared access to mounted volumes, so we introduced a way to remove this uniqueness check:

Reviewer: ```suggestion
allow shared access to mounted volumes, so we introduced a way to disable the uniqueness check:
```

> There is a new field, `container.volumes[n].external.shared` which defaults to `false`. If set to true, the same volume name can be used

Reviewer: ```suggestion
A new field, `container.volumes[n].external.shared` which defaults to `false`. If set to true, the same volume name can be used
```
